libs/ml/include/ml/estimators/dnn_classifier.hpp

```diff
@@ -203,9 +219,17 @@ bool DNNClassifier<TensorType>::Predict(TensorType &input, TensorType &output)
 }
 
 template <typename TensorType>
-void DNNClassifier<TensorType>::PrintStats(SizeType epoch, DataType loss)
+void DNNClassifier<TensorType>::PrintStats(SizeType epoch, DataType loss, DataType val_loss)
 {
-  std::cout << "epoch: " << epoch << ", loss: " << loss << std::endl;
+  if (val_loss == fetch::math::numeric_max<DataType>())
```

Comment:
wouldn't it be better for this if/else to be a switch/case based on current mode_ enum?